sstables/mx/partition_reversing_data_source.cc

```diff
@@ -157,16 +157,17 @@ class row_body_skipping_context : public data_consumer::continuous_data_consumer
 
     void setup_ck(const std::vector<std::optional<uint32_t>>& column_value_fix_lengths) {
         if (column_value_fix_lengths.empty()) {
-            _ck_column_value_fix_lengths = boost::make_iterator_range(column_value_fix_lengths);
+            _ck_column_value_fix_lengths = std::ranges::subrange(column_value_fix_lengths);
         } else {
-            _ck_column_value_fix_lengths = boost::make_iterator_range(std::begin(column_value_fix_lengths),
-                                                                      std::begin(column_value_fix_lengths) + _ck_size);
+            _ck_column_value_fix_lengths = std::ranges::subrange(std::begin(column_value_fix_lengths),
+                                                                 std::begin(column_value_fix_lengths) + _ck_size);
         }
         _ck_blocks_header_offset = 0u;
     }
     bool no_more_ck_blocks() const { return _ck_column_value_fix_lengths.empty(); }
     void move_to_next_ck_block() {
-        _ck_column_value_fix_lengths.advance_begin(1);
+        _ck_column_value_fix_lengths = std::ranges::subrange(_ck_column_value_fix_lengths.begin() + 1,
+                                                             _ck_column_value_fix_lengths.end());
```

Comment:
There is subrange::advance(), but it doesn't matter too much.

Reply:
TIL. will send a follow up later on. thanks!